src/d3a/d3a_core/sim_results/stats.py

```diff
@@ -226,36 +227,40 @@ def _aggregate_totals_with_penalties(self, area, results):
         results[area.name]["totals_with_penalties"] = {"costs": totals_with_penalties}
 
     def _generate_external_and_total_bills(self, area, results, flattened):
+        if area.name in flattened:
+            # External trades are the trades of the parent area
+            external = flattened[area.name].copy()
+            # Should not include market fee to the external trades
+            market_fee = external.pop("market_fee", 0.)
+            # Should switch spent/earned and bought/sold, to match the perspective of the UI
+            spent = external.pop("spent") + market_fee
+            earned = external.pop("earned")
+            bought = external.pop("bought")
+            sold = external.pop("sold")
+            external.update(**{
+                "spent": earned, "earned": spent, "bought": sold,
+                "sold": bought, "market_fee": 0})
+            external["total_energy"] = external["bought"] - external["sold"]
+            external["total_cost"] = external["spent"] - external["earned"]
+            results[area.name].update({"External Trades": external})
+
         all_child_results = [v for v in results[area.name].values()]
         results[area.name].update({"Accumulated Trades": {
             'bought': sum(v['bought'] for v in all_child_results),
             'sold': sum(v['sold'] for v in all_child_results),
             'spent': sum(v['spent'] for v in all_child_results),
-            'earned': sum(v['earned'] for v in all_child_results),
+            'earned': sum(v['earned'] for v in all_child_results) + self.market_fees[area.name],
```

Comment:
I think this is not correct here. I think the earned value of the Accumulated Trades should not include the market fees (with an emphasis on 'I think', I might have misunderstood what needs to be shown here)

Reply:
Resolved after offline discussion, no need for change